### ISSUE TITLE
Sessions expire in loadbalanced application

### ISSUE THREAD

**Issue by Marcus-Friis** [group-o-minitwit-2024/MiniTwit#69]
When using MiniTwit, since it is not entirely stateless, you are logged out whenever you are directed to a new loadbalanced server. One possible solution for this could be to use an `ip_hash` load balancing strategy:
> [IP Hash](https://nginx.org/en/docs/http/ngx_http_upstream_module.html#ip_hash) – The server to which a request is sent is determined from the client IP address. In this case, either the first three octets of the IPv4 address or the whole IPv6 address are used to calculate the hash value. The method guarantees that requests from the same address get to the same server unless it is not available.

Alternatively, we could use `sticky cookies`
> [Sticky cookie](https://nginx.org/en/docs/http/ngx_http_upstream_module.html#sticky_cookie) – NGINX Plus adds a session cookie to the first response from the upstream group and identifies the server that sent the response. The client’s next request contains the cookie value and NGINX Plus route the request to the upstream server that responded to the first request.

**Comment by laugekj** [comment]
Yes, and this sadly courses users to lose their session resulting in them being logged out of the application :(
